autogen/agentchat/contrib/agent_builder.py

```diff
@@ -519,6 +568,17 @@ def build_from_library(
             )
         build_manager = autogen.OpenAIWrapper(config_list=config_list)
 
+        agent_model_config_list = autogen.config_list_from_json(
+            self.config_file_or_env,
+            file_location=self.config_file_location,
+            filter_dict={"model": self.agent_model},
+        )
+        if len(agent_model_config_list) == 0:
+            raise RuntimeError(
+                f"Fail to initialize agent model: {self.agent_model} does not exist in {self.config_file_or_env}. "
+                f'If you want to change this model list, please specify the "agent_model" in the constructor.'
+            )
```

Comment:
Could we extract this part and  line 559-568 into a config checking function to  reduce duplication in `build` and  `build_from_library`?

Reply:
Done